tests/train_valid_test_split.R

```diff
@@ -0,0 +1,77 @@
+#' train_valid_test_split
+#' Splits arrays or matrices into random train, validation and test subsets
+#' The proprotion of the train set relative to the input data will be valid_size * (1 - test_size)
+#'
+#' @param
+#' X, y arrays
+#' @param X (data frame)
+#' Original features from data set
+#' @param y (data frame)
+#' Original labels from data set
+#' @param test_size (double, default = NULL)
+#' Value between 0.0 and 1.0 to represent the proportion of the dataset to comprise the size of the test subset
+#' @param valid_size float or None (default = 0.25)
+#' Value between 0.0 and 1.0 to represent the proportion of the dataset to comprise the size of the test subset
+#' @param random_state (int, default = NULL)
+#' Seed for the random number generator
+#' @shuffle (default = TRUE)
+#' Shuffle the rows or not
+#'
+#' @import gensvm
+#'
+#' @return
+#' @export
+#'
+#' @examples
+#' x <- iris[, -5]
+#' y <- iris[, 5]
+#'
+#' Use default values
+#' split <- gensvm.train.test.split(x, y)
+
+train_valid_test_split <- function(X,
+                                   y = NULL,
+                                   test_size = 0.25,
+                                   valid_size = 0.25,
+                                   shuffle = TRUE,
+                                   random_state = NULL) {
+
+  #' Check data input types
+  if (class(x) != 'data.frame') stop('You have not inputted an acceptable data type for x')
```

Comment:
class(x): x should be capital letter as per input parameter
```suggestion
  if (class(X) != 'data.frame') stop('You have not inputted an acceptable data type for X')
```